version_info.go

```diff
@@ -0,0 +1,306 @@
+		uintptr(len(info)),
+		uintptr(unsafe.Pointer(&info[0])),
+	)
+	if ret == 0 {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo from windows; %s", err)
+	}
+
+	vi := Info{data: info}
+
+	fixedInfo := vi.GetFixedInfo()
+	vi.FixedFileInfo = fixedInfo
+
+	return vi, nil
+}
+
+// GetPageID gets pageID filed in structure.
+// It tries to get pageID from VersionInfo data.
+// If getting ends with fail, it returns default pageID
+// 040904E4 // US English + CP_USASCII.
+func (f Info) GetPageID() Locale {
+	data, err := f.verQueryValue(`\VarFileInfo\Translation`, false)
+	if err != nil || len(data) == 0 {
+		return EnglishUSAscii
+	}
+
+	// each pageID consists of a 16-bit language ID and a 16-bit code page
+	type langAndCodePage struct {
+		Language uint16
+		CodePage uint16
+	}
+	if len(data)%int(unsafe.Sizeof(langAndCodePage{})) != 0 {
+		return EnglishUSAscii
+	}
+	n := len(data) / int(unsafe.Sizeof(langAndCodePage{}))
+	if n == 0 {
+		return EnglishUSAscii
+	}
+
+	ids := (*[1 << 28]langAndCodePage)(unsafe.Pointer(&data[0]))[:n:n]
+	return Locale(fmt.Sprintf("%04x%04x", ids[0].Language, ids[0].CodePage))
+}
+
+// GetFixedInfo returns FixedFileInfo structure, which constructs from windows
+// VS_FIXEDFILEINFO structure.
+// source:
+// https://helloacm.com/c-function-to-get-file-version-using-win32-api-ansi-and-unicode-version/
+func (f Info) GetFixedInfo() FixedFileInfo {
+	data, err := f.verQueryValue(`\`, false)
+	if err != nil {
+		return FixedFileInfo{}
+	}
+	// source:
+	// https://docs.microsoft.com/en-us/windows/win32/api/verrsrc/ns-verrsrc-vs_fixedfileinfo
+	type rawFixedFileInfo struct {
+		Signature        uint32
+		StrucVersion     uint32
+		FileVersionMS    uint32
+		FileVersionLS    uint32
+		ProductVersionMS uint32
+		ProductVersionLS uint32
+		FileFlagsMask    uint32
+		FileFlags        uint32
+		FileOS           uint32
+		FileType         uint32
+		FileSubtype      uint32
+		FileDateMS       uint32
+		FileDateLS       uint32
+	}
+	vsFixedInfo := *((*rawFixedFileInfo)(unsafe.Pointer(&data[0])))
+	return FixedFileInfo{
+		FileMajor:      uint16(vsFixedInfo.FileVersionMS >> 16),
+		FileMinor:      uint16(vsFixedInfo.FileVersionMS & 0xffff),
+		FileBuild:      uint16(vsFixedInfo.FileVersionLS >> 16),
+		FilePrivate:    uint16(vsFixedInfo.FileVersionLS & 0xffff),
+		ProductMajor:   uint16(vsFixedInfo.ProductVersionMS >> 16),
+		ProductMinor:   uint16(vsFixedInfo.ProductVersionMS & 0xffff),
+		ProductBuild:   uint16(vsFixedInfo.ProductVersionLS >> 16),
+		ProductPrivate: uint16(vsFixedInfo.ProductVersionLS & 0xffff),
+		FileFlagsMask:  vsFixedInfo.FileFlagsMask,
+		FileFlags:      vsFixedInfo.FileFlags,
+		FileOs:         vsFixedInfo.FileOS,
+		FileType:       vsFixedInfo.FileType,
+		FileSubType:    vsFixedInfo.FileSubtype,
+		FileDateMS:     vsFixedInfo.FileDateMS,
+		FileDateLS:     vsFixedInfo.FileDateLS,
+	}
+}
+
+// GetProperty returns string-property.
+func (f Info) GetProperty(propertyName string) (property string) {
```

Comment:
I think it should be additionally noted that if `GetProperty` failed to query a property within the locale set for `Info` it will try to query it with default locales( `locale. EnglishUSAscii` ...)

\+ I suppose we need an additional func to get the property with the given translation without any heuristics